python/lsst/summit/utils/peekExposure.py

```diff
@@ -0,0 +1,1026 @@
+        For displaying the exposure and sources.
+
+    Notes
+    -----
+    The basic philosophy of PeekExposureTask is to:
+    1) Classify exposures based on metadata into 'donut', 'spec', and 'photo'.
+    2) Run CharacterizeImageTask on the exposure, but with modifications to run
+       as fast as reasonably possible (e.g., turning features off and binning
+       the image), and also some class specific modifications.
+    3) Return a source catalog and some summary statistics.
+    4) Try only to branch in the code based on the metadata, and not on the
+       data itself.  This avoids problematic discontinuities in summary
+       statistics.
+
+    For speed, many items in CharacterizeImageTask have been turned off,
+    including:
+        - PSF measurement (though not moment measurements)
+        - Deblending
+        - Repair
+        - Aperture correction
+        - Background estimation
+        - Noise replacement
+        - Writing to disk
+
+    The main knobs we fiddle with based on the classification are:
+        - Detection threshold
+        - Minimum number of pixels for a detection
+        - Binning of the image
+        - Installed PSF size
+    """
+
+    ConfigClass = PeekExposureTaskConfig
+    _DefaultName = "peekExposureTask"
+
+    def __init__(self, config, *, display=None, **kwargs):
+        super().__init__(config=config, **kwargs)
+
+        self.makeSubtask("donut")
+        self.makeSubtask("photo")
+        self.makeSubtask("spec")
+
+        self._display = display
+
+    def getDonutDiameter(self, exposure):
+        """Estimate donut diameter from exposure metadata.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure to estimate donut diameter for.
+
+        Returns
+        -------
+        donutDiameter : float
+            Estimated donut diameter in pixels.
+        """
+        visitInfo = exposure.getInfo().getVisitInfo()
+        focusZ = visitInfo.focusZ
+        instrumentLabel = visitInfo.instrumentLabel
+
+        match instrumentLabel:
+            case "LATISS":
+                focusZ *= 41  # magnification factor
+                fratio = 18.0
+            case "LSSTCam" | "ComCam":
+                fratio = 1.234
+        # AuxTel/ComCam/LSSTCam all have 10 micron pixels (= 10e-3 mm)
+        donutDiameter = abs(focusZ) / fratio / 10e-3
+        self.log.info(f"{focusZ=} mm")
+        self.log.info(f"donutDiameter = {donutDiameter} pixels")
+        return donutDiameter
+
+    def run(
+        self,
+        exposure,
+        *,
+        doDisplay=False,
+        doDisplayIndices=False,
+        mode="auto",
+        donutDiameter=None,
+    ):
+        """
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure at which to peek.
+        doDisplay : bool, optional
+            Display the exposure and sources?  Default False.
```

Comment:
Maybe a tad picky, but might be worth noting here that a display is required to have been pass on init for this option.